usl_pipeline/dev-requirements.txt

```diff
@@ -4,10 +4,12 @@ flake8==7.0.0
 iniconfig==2.0.0
 mccabe==0.7.0
 mypy-extensions==1.0.0
+numpy==1.26.4
```

Comment:
We want to include the full set of dependencies -- the full transitive closure -- in the requirements file.
We want to have just top-level dependencies, like numpy and rasterio, in setup.py
I showed how to do this over here:
https://github.com/UrbanSystemsLab/climateiq-cnn/pull/6/files

You can use `pip freeze > requirements.txt` to dump the full set of installed packages.

We want to state the full set of dependencies in a requirements file to avoid pulling in different versions of the dependencies-of-dependencies every time we pip install them.

The requirements in setup.py are meant to be looser so it's easier to install as one of many libraries which might require more certain versions of various packages.

Reply:
Done